plugins/automox/icon_automox/util/api_client.py

```diff
@@ -0,0 +1,318 @@
+        }
+
+    def _call_api(self, method: str, url: str, params: Dict = dict(), json_data: object = None) -> json:
+        try:
+            response = self.session.request(method, url, json=json_data, params=params)
+
+            if response.status_code == 403:
+                raise PluginException(preset=PluginException.Preset.API_KEY)
+
+            if response.status_code == 404:
+                raise PluginException(preset=PluginException.Preset.NOT_FOUND)
+
+            if response.status_code == 204:
+                return
+
+            if 200 <= response.status_code < 300:
+                return response.json()
+
+            # Non-success and unknown
+            raise PluginException(cause=f"An error occurred when making {method} request to {url}.",
+                                  assistance=f"Response code: {response.status_code}, Content: {response.text}.")
+        except json.decoder.JSONDecodeError as e:
+            self.logger.info(f"Invalid json: {e}")
+            raise PluginException(preset=PluginException.Preset.INVALID_JSON)
+        except requests.exceptions.HTTPError as e:
+            self.logger.info(f"Call to Automox Console API failed: {e}")
+            raise PluginException(preset=PluginException.Preset.UNKNOWN)
+
+    def _page_results(self, url: str, init_params: Dict = dict(), sanitize: bool = True) -> json:
+        params = self.first_page(init_params)
+
+        page_resp = []
+
+        while True:
+            resp = self._call_api("get", url, params)
+            if sanitize:
+                resp = self.remove_null_values(resp)
+
+            page_resp.extend(resp)
+
+            self.logger.info(f"Page {params.get('page')} result count: {len(resp)}")
+
+            if len(resp) < self.PAGE_SIZE:
+                break
+
+            self.next_page(params)
+
+        return page_resp
+
+    def _page_results_data(self, url: str, init_params: Dict = dict()) -> json:
+        params = self.first_page(init_params)
+
+        page_resp = []
+
+        while True:
+            resp = self._call_api("get", url, params)
+            resp_data = self.remove_null_values(resp.get('data'))
+
+            page_resp.extend(resp_data)
+
+            self.logger.info(f"Page {params.get('page')} result count: {len(resp_data)}")
+
+            if len(resp_data) < self.PAGE_SIZE:
+                break
+
+            self.next_page(params)
+
+        return page_resp
+
+    # Remove Null from response to avoid type issues
+    def remove_null_values(self, d):
+        if type(d) is dict:
+            return dict((k, self.remove_null_values(v)) for k, v in d.items() if v and self.remove_null_values(v))
+        elif type(d) is list:
+            return [self.remove_null_values(v) for v in d if v and self.remove_null_values(v)]
+        else:
+            return d
+
+    @staticmethod
+    def _org_param(org_id):
+        return {"o": org_id}
+
+    @staticmethod
+    def first_page(params: Dict = {}):
```

Comment:
```suggestion
    def first_page(params: Dict = {}) -> None:
```
Depending on if you accept the below change suggestion, you may need to change this return type.